### ISSUE TITLE
Implement bridge loop detection

### ISSUE THREAD

**Issue by ioolkos** [erlio/vernemq#753]
It's been long known that VerneMQ does not implement automatic loop detection for bridges like
`vmq_bridge.tcp.br0.topic.1 = bbb both 1`

Luckily this config still works bridging from VerneMQ to brokers that do have loop detection (like Mosquitto).
It does not work bridging 2 VerneMQ brokers though... so this is kind of hard to understand and differentiate.

Between 2 VerneMQ brokers you currently have to separate pub and sub topics by using local-prefix and remote-prefix like this:

`vmq_bridge.tcp.br0.topic.1 = bbb both 1 B1 B2`

Subscribe:
On broker B1: subscribe to B1/bbb
On broker B2: subscribe to B2/bbb

Publish:
On broker B1: publish to bbb -> will publish message on B2/bbb
On broker B2: publish to bbb -> will publish message on B1/bbb

Conclusion: implementing loop detection might be a good thing after all :)

**Comment by bsninh78** [comment]
I have the same issue for loop messages when I configure bridge for 2 VerneMQ as @ioolkos 
How I can stop loop for the bridge?

**Comment by codeadict** [comment]
Any idea of how can this be reliably implemented @ioolkos? Probably the `try_private` setting right? Or maybe sending a canary message and check for a feedback loop?

**Comment by ioolkos** [comment]
Cool. 
I forgot what the reason was that this is not yet implemented.
If we look at the vmq_bridge schema, we can name and start multiple bridges. Each bridge will have an TCP endpoint, so it will be within same client session context in the bridge, so we can probably detect by inspecting for some try_private bit whether the message comes from ourselves (from bridge perspective).
I'm currently not 100% sure whether this is possible for 2 different bridges.

**Comment by codeadict** [comment]
Yeah, this may be fine for a VerneneMQ to verneMQ bridge, `try_private` is not supported by all brokers, unfortunately. I'll go ahead and start implementing it for VerneMQ bridges

**Comment by ioolkos** [comment]
Note that for VerneMQ bridges I didn't yet manage to fix https://github.com/vernemq/vernemq/issues/1420, that is handling the retain flag correctly within VerneMQ <-> VerneMQ bridges.
I think it's unrelated to loop detection though, but maybe you'll encounter the same issues/questions.

**Comment by codeadict** [comment]
Thanks for the heads up. That would be fun :)
